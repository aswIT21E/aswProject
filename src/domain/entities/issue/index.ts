export * from "./issue";
export * from "./issue.interface";
export * from "./issue.schema";
