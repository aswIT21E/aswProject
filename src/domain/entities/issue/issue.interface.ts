import type { IComment, IUser } from '~/domain/entities';
import { MongoId } from '~/types/types';

import type { Activity, IActivity } from '../activity';

export interface IIssue {
  id: MongoId;
  numberIssue: number;
  subject: string;
  description: string;
  activity: Activity[];
  creator: IUser; // FK a IUser
  status: string;
  severity: string;
  type: string;
  date: string;
  priority: string;
  comments?: IComment[];
  locked: boolean;
  watchers: IUser[];

  lockIssue: () => void;
  unlockIssue: () => void;
  updateWatchers: (watchers: IUser[]) => void;
  addActivity: (activity: IActivity) => void;
  readonly watchersIds: MongoId[];
  readonly activitiesIds: MongoId[];
}
