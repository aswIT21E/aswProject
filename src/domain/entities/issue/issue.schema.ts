import { Schema, model } from 'mongoose';

import type { IIssue } from './issue.interface';

const IssueSchema = new Schema({
  subject: { required: true, type: String },
  numberIssue: { required: true, type: Number },
  description: { required: true, type: String },
  creator: { required: true, type: Schema.Types.ObjectId, model: 'User' }, // FK a IUser
  status: { required: true, type: String },
  type: { required: true, type: String },
  date: { required: false, type: String },
  severity: { required: true, type: String },
  priority: { required: true, type: String },
  comments: [{ required: true, type: Schema.Types.ObjectId, model: 'Comment' }],
  locked: { required: true, type: Boolean, default: false },
  watchers: [{ required: true, type: Schema.Types.ObjectId, model: 'User' }],
});

const IssueModel = model<IIssue>('Issue', IssueSchema);

export { IssueModel };
