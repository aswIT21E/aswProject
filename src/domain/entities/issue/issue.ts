import { IComment } from '../comment';
import type { IIssue } from './issue.interface';

export class Issue implements IIssue {
  public id: string;
  public numberIssue: number;
  public subject: string;
  public description: string;
  public creator: string; // FK a IUser
  public status: string;
  public type: string;
  public severity: string;
  public priority: string;
  public date: string;
  public comments?: IComment[];
  public locked: boolean;

  constructor(
    id: string,
    numberIssue: number,
    subject: string,
    description: string,
    creator: string,
    status: string,
    type: string,
    severity: string,
    priority: string,
    comments: IComment[],
  ) {
    this.id = id;
    this.numberIssue = numberIssue;
    this.subject = subject;
    this.description = description;
    this.creator = creator;
    this.status = status;
    this.type = type;
    this.severity = severity;
    this.priority = priority;
    this.comments = comments;
    this.locked = false;
  }
}
