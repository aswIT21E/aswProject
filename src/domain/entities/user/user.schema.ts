import { Schema, model } from 'mongoose';

import type { IUser } from './user.interface';

const UserSchema = new Schema({
  email: { required: true, type: String },
  name: { required: true, type: String },
  username: { required: true, type: String },
  password: { required: true, type: String },
  bio: { required: true, type: String },
  profilePicture: { required: true, type: String },
});

const UserModel = model<IUser>('User', UserSchema);

export { UserModel };
