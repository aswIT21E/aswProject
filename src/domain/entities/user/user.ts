import type { MongoId } from '~/types/types';

import type { IUser } from './user.interface';

export class User implements IUser {
  public id: MongoId;
  public email: string;
  public name: string;
  public username: string;
  public password: string;
  public bio: string;

  constructor(id: MongoId, email: string, name: string, username: string, password: string, bio: string) {
    this.id = id;
    this.email = email;
    this.name = name;
    this.username = username;
    this.password = password;
    this.bio = bio;
  }
}
