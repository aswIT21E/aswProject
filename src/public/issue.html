<link rel="stylesheet" type="text/css" href="issue.css">


  <div class="issue">
    <div class="bola" id="TipoBola"></div>
    <div class="bola" id="GravedadBola"></div>
    <div class="bola" id="PrioridadBola"></div>
    <div class="informacion">
      <div class="numero-peticion" id="NumPeticion"></div>
      <div class="texto-peticion" id="TextoPeticion"><a id="linkIssue" href="localhost:8080/issues/issue=000">Node 12 is end-of-life aaaaaaaaaa</a> </div>
    </div>
    <div class="estado" id= "EstadoPeticion"></div>
    <div class="fecha-creacion" id = "FechaPeticion"></div>
  </div>
  

  