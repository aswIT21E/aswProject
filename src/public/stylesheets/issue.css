/* Estilos para el botón de filtro */
.filter-button {
  display: flex;
  align-items: center;
  background-color: #f2f2f2;
  border: none;
  color: #008aa8;
  font-size: 14px;
  padding: 10px 15px;
  text-align: center;
  cursor: pointer;
  border-radius: 4px;
}

.filter-button:hover {
  background-color: #ddd;
}
.new-issue-button {
  justify-content: right;
  background-color: #83eede;
  border: none;
  color: #2e3440;
  font-size: 14px;
  padding: 10px 15px;
  text-align: right;
  cursor: pointer;
  border-radius: 4px;
  position: absolute;
  right:5%;
}
.new-issue-button:hover {
  background-color: #ddd;
}

/* Estilos para el icono de filtro */
.fa-filter {
  margin-right: 5px;
}

/* Estilos para la caja de búsqueda */
.search-box {
  display: flex;
  align-items: center;
  background-color: #f2f2f2;
  border-radius: 4px;
  width: 250px;
  margin: 0 20px;
}

.search-filter {
  justify-content: start;
  display: flex;
  min-width: 500px;
}
/* Estilos para el icono de búsqueda */
.fa-search {
  margin: 0 10px;
}

/* Estilos para la entrada de búsqueda */
.search-input {
  border: none;
  background-color: #f2f2f2;
  padding: 10px;
  font-size: 14px;
  width: 100%;
}

/* Estilos para TAGS*/
.switch {
  display: flex;
  position: relative;
  width: 40px;
  height: 20px;
  margin: 10px;
}

.switch input {
  display: flex;
}
.switch label {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: grey;
  border-radius: 34px;
  transition: background-color 0.3s;
}

.switch label:before {
  position: absolute;
  content: "";
  height: 16px;
  width: 16px;
  left: 2px;
  bottom: 2px;
  background-color: white;
  border-radius: 50%;
  transition: transform 0.3s;
}

.switch input:checked + label {
  background-color: #008aa8;
}
.label {
  margin-left: 30px;
  font-size: 16px;
  font-weight: 400;
  align-items: center;
  font-size: 16px;
  justify-content: center;
}
.switch input:checked + label:before {
  transform: translateX(20px);
}

.new-issue-button {
  justify-content: end;
}
