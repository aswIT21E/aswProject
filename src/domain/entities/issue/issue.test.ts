import { Types } from 'mongoose';
import { User } from '../user';

import { Issue } from './issue';
import type { IIssue } from './issue.interface';

describe('Issue Entity', function () {
  let instance: IIssue;

  beforeEach(function () {
    instance = new Issue(
      new Types.ObjectId(),
      1234,
      'test-subject',
      'test-description',
      new User(
        new Types.ObjectId(),
        'test-email',
        'test-username',
        'test-password',
      ),
      'done',
      'bug',
      '2/2/2002',
      'high',
      'high',
      [],
    );
  });

  it('can be created', function () {
    expect(instance).toMatchSnapshot();
  });

  it('locksIssue sets locked property as true', function () {
    instance.lockIssue();

    expect(instance.locked).toBeTruthy();
  });

  it('unlocksIssue sets locked property as false', function () {
    instance.unlockIssue();

    expect(instance.locked).toBeFalsy();
  });
});
