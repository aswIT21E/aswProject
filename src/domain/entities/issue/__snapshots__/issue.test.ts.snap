// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue Entity can be created 1`] = `
Issue {
  "creator": "test-creator",
  "description": "test-description",
  "id": "test-id",
  "priority": "high",
  "severity": "high",
  "status": "done",
  "subject": "test-subject",
  "type": "bug",
}
`;
